src/Disks/ObjectStorages/VFSTransactionGroup.cpp

```diff
@@ -0,0 +1,36 @@
+#include "VFSTransactionGroup.h"
+#include "DiskObjectStorageVFS.h"
+
+namespace DB
+{
+VFSTransactionGroup::VFSTransactionGroup(DiskPtr disk_)
+    : disk(disk_->isObjectStorageVFS() ? static_cast<DiskObjectStorageVFS *>(disk_.get()) : nullptr)
+{
+    if (!disk)
+        return;
+    if (disk->tryAdd(this))
+        LOG_TRACE(disk->log, "TransactionGroup: set");
+    else
+    {
+        LOG_TRACE(disk->log, "TransactionGroup: failed to set");
+        disk = nullptr;
+    }
+}
+
+VFSTransactionGroup::~VFSTransactionGroup()
+{
+    if (!disk || std::uncaught_exceptions())
```

Comment:
Incorrect checking of unwinding. See example in https://en.cppreference.com/w/cpp/error/uncaught_exception